### ISSUE TITLE
Packaging the API

### ISSUE THREAD

**Issue by taconaut** [Omertron/api-thetvdb#18]
Hi,

Since updating to version 1.8 of your library I've got issues packaging a working jar and hoped you could point me in the right direction.
I use the library in a plugin for [ums-mlx](https://github.com/taconaut/ums-mlx) where the plugin is being dynamically loaded on runtime. As I got a `java.lang.NoClassDefFoundError` when instantiating the library with `private TheTVDBApi theTvDbApi = new TheTVDBApi("secret");`. This led me to think either a dependency is missing or included with the wrong version. That's why I've included all the dependencies with the version specified in the [maven repository](http://mvnrepository.com/artifact/com.omertron/thetvdbapi/1.8) in the project and package them in the jar with [this pom.xml](https://github.com/taconaut/ums-mlx/blob/6d575d673e0f6ef2e7a86fd2b8b59e6122604bd1/plugins/TheTVDBEpisodeImportPlugin/pom.xml).

Now I get this exception:

```
WARN  14:17:10.730 [net.pms.PMS.main()] The plugin 'net.pms.plugin.fileimport.TheTVDBImportPlugin' couldn't be loaded
java.lang.NoSuchFieldError: INSTANCE
    at org.apache.http.conn.ssl.SSLConnectionSocketFactory.<clinit>(SSLConnectionSocketFactory.java:144)
    at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:955)
    at org.yamj.api.common.http.SimpleHttpClientBuilder.build(SimpleHttpClientBuilder.java:208)
    at com.omertron.thetvdbapi.TheTVDBApi.<init>(TheTVDBApi.java:68)
    at net.pms.plugin.fileimport.TheTVDBImportPlugin.<init>(TheTVDBImportPlugin.java:72)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
    at java.lang.Class.newInstance(Class.java:442)
    at net.pms.plugins.PluginsFactory.lookup(PluginsFactory.java:323)
    at net.pms.PMS.init(PMS.java:579)
    at net.pms.PMS.createInstance(PMS.java:1112)
    at net.pms.PMS.main(PMS.java:1281)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:297)
    at java.lang.Thread.run(Thread.java:745)
```

The exception still originates from the instantiation of `TheTVDBApi`.

Would you happen to know what I am missing?

**Comment by Omertron** [comment]
Looks like the problem is originating in the api-common library ( https://github.com/YAMJ/api-common )

can you check any dependencies with in that? I will check from my end that they are all the same versions.

**Comment by taconaut** [comment]
After a long pause I've started tackling this issue again and found a dirty solution to it.

First, the issue is not related to your library. When I run my unit tests everything works fine; also the instantiation which caused the above described issue. When dynamically instantiating my plugin through ums-mlx the exception occurs.
Thus, it must be a dependency issue. There are references to e.g. HttpClient in ums-mlx which require another version than the plugin.

If I instantiate the library with the deprecated `DefaultHttpClient`: `theTvDbApi = new TheTVDBApi("---", new DefaultHttpClient())` everything works fine.

I've also tried to properly use `HttpClientBuilder` with `HttpClientBuilder.create().build()` and included all dependencies in the required version in the plugin and checked where the classes are being loaded from. Everything was looking fine to me but the excpetion remained. I'm going to keep using `DefaultHttpClient`.
